### DISCUSSION TITLE
Generate migrations for deployment and dev with one EntityManager

### DISCUSSION THREAD

**Discussion by daviddavo** [symfony/symfony#49633] [Q&A] [answered]
I generate my migrations on local for testing, where I use a sqlite DB on `var/data.db`. When I generate the migrations (just for testing), it creates a file in `src/Migrations` that checks if the engine is also `sqli`. When I try to apply the migrations on the deployment environment, it tells me that:

```
Migration 20220227181934 failed during Execution. Error Migration can only be executed safely on 'sqlite'.
```

The env variable `DATABASE_URL` changes between local and prod.

How could I solve this issue? What approaches do I have? Should I just skip uploading the `src/Migrations` folder to the server and let it handle it? Or should I generate migrations also for mysql/postgres? How?

> Note: Similar discussions talk about using multiple EntityManagers, but I just have one

**Comment by ghost** [comment, marked as answer]
Just remove the autogenerated line `$this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'sqlite', 'Migration can only be executed safely on \'sqlite\'.');`. It's useless as everybody working on a project should be aware that prod and test environments are running different database vendor and as a consequence should write migrations with SQLs that are understood by both vendors.
